core/node/registries/river_registry_contract.go

```diff
@@ -393,13 +423,133 @@ func (c *RiverRegistryContract) ForAllStreams(
 			if err != nil {
 				return err
 			}
+			totalStreams++
 			if !cb(makeGetStreamResult(streamId, &stream.Stream)) {
 				return nil
 			}
 		}
 	}
 
-	log.Info("RiverRegistryContract: GetPaginatedStreams completed", "elapsed", time.Since(startTime))
+	elapsed := time.Since(startTime)
+	log.Info(
+		"RiverRegistryContract: GetPaginatedStreams completed",
+		"elapsed",
+		elapsed,
+		"streamsPerSecond",
+		float64(totalStreams)/elapsed.Seconds(),
+	)
+
+	return nil
+}
+
+// ForAllStreams calls the given cb for all streams that are registered in the river registry at the given block num.
+// If cb returns false forAllStreams returns.
+func (c *RiverRegistryContract) forAllStreamsParallel(
+	ctx context.Context,
+	blockNum crypto.BlockNumber,
+	cb func(*GetStreamResult) bool,
+) error {
+	log := dlog.FromCtx(ctx)
+
+	numWorkers := c.Settings.ParallelReaders
+	if numWorkers <= 1 {
```

Comment:
I don't think this will ever be hit, b/c of https://github.com/river-build/river/pull/1516/files#diff-15b39610ef19b415cd3d948a79ebe97f7c9fcee7a2cb83316a584289057b6d2fR367.

Reply:
this handles the case if setting was set explicitly